sycl/test-e2e/Basic/intel-ext-device.cpp

```diff
@@ -3,7 +3,6 @@
 // RUN: env ONEAPI_DEVICE_SELECTOR=opencl:gpu %t.out
 //
 // REQUIRES: gpu
```

Comment:
Since all the listed intel-extensions in this test were implemented for all the other backends in addition to L0, I was able to verify this test manually by compiling the file for CUDA and HIP and was hoping it might makes sense not to just restrict it to L0-only.

Reply:
Are you saying that `dev.get_info<ext::intel::info::device::pci_address>()` returns something meaningful on non-Intel devices?
I thought that intel vendor extensions are supposed to support only Intel devices. 
@gmlueck, aren't we need oneapi extension in this case?

Reply:
My expectation is that all aspects in this test are specific to Intel devices and all checks must be skipped on HIP and CUDA back-ends. I would expect `ext_intel_free_memory` to be "promoted" to `ext_oneapi_free_memory` before it's implemented for non-intel devices. 

From https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/README-process.md#creating-a-new-extension:

> For DPC++, we use the string "oneapi" unless the extension is very specific to Intel hardware in which case we use the string "intel".

Reply:
perfect, thanks for pointing the documentation. 

Could you confirm if `REQUIRES: level_zero` is an appropriate one, to be replaced for these lines:
```bash
// REQUIRES: gpu
// UNSUPPORTED: hip
// Temporarily disable on L0 due to fails in CI
```

Reply:
I looked through the file and found a few inconsistencies. I think someone from the @intel/llvm-reviewers-runtime should take a look.

The comment says that this test is supposed to executed only on Level Zero back-end, but we run it on OpenCL back-end too. The test validates Intel GPU specific features and Intel OpenCL GPU potentially can expose these API too. @intel/llvm-reviewers-runtime, is `intel_device_info` extension supported by OpenCL back-end? If yes, we must update the comment, otherwise, update run command to mark OpenCL back-end as unsupported.

In addition, the test validates `oneapi_max_work_group_query` extension and this extension can be supported by HIP and CUDA backends. This part of the test should be moved to a separate file and executed on all supported back-ends.

@aelovikov-intel, I think you wanted to provide an infrastructure to set aspect-based requirements in the tests. With that infrastructure in place, we should use something like this:
```
// REQUIRES: intel_device_info
// RUN: %GPU_RUN_PLACEHOLDER %t.out
```

In case don't have it yet.

```
// REQUIRES: level_zero || opencl
// RUN: %GPU_RUN_PLACEHOLDER %t.out
```

Reply:
> @aelovikov-intel, I think you wanted to provide an infrastructure to set aspect-based requirements in the tests. 

That hasn't been implemented yet.

Reply:
> My expectation is that all aspects in this test are specific to Intel devices and all checks must be skipped on HIP and CUDA back-ends.

It is valid to check the aspects like `aspect::ext_intel_free_memory` on any device, even non-Intel devices.  The strict requirement is that the aspect should only be true if the associated info descriptor (e.g. `info::device::global_mem_size`) is supported.  Therefore, code like this:

```
if (dev.has(aspect::ext_intel_free_memory)) {
  auto TotalMemory = dev.get_info<info::device::global_mem_size>();
  /* test "TotalMemory" */
```

is safe to call on any device.  I would expect the aspect check to normally return false for non-Intel devices, but that is not a strict requirement.  The strict requirement is only that the aspect may return true only when the info descriptor is supported.

Our convention is to name extensions with "intel" if we think they are useful only in Intel devices.  If we later decide that an API is useful on other devices, we should consider adding a "oneapi" version.

Reply:
> The strict requirement is that the aspect should only be true if the associated info descriptor (e.g. `info::device::global_mem_size`) is supported.

Do you mean `ext::intel::info::device::free_memory` info descriptor? I don't see how `info::device::global_mem_size` is associated with `aspect::ext_intel_free_memory`.

> It is valid to check the aspects like `aspect::ext_intel_free_memory` on any device, even non-Intel devices.

> If we later decide that an API is useful on other devices, we should consider adding a "oneapi" version.

I'm confused. It looks like we put an API useful on other devices to the extension which is supposed to be useful only in Intel devices. Right?

Reply:
> Do you mean `ext::intel::info::device::free_memory` info descriptor? I don't see how `info::device::global_mem_size` is associated with `aspect::ext_intel_free_memory`.

Yes.  Sorry, I copy-pasted the wrong thing.

> I'm confused. It looks like we put an API useful on other devices to the extension which is supposed to be useful only in Intel devices. Right?

It was added specifically for Level Zero.  I'm actually not sure if this is generally useful.